### PR TITLE
Upgrade lz4 to v3, add unittests

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+OTEL_DOCKER_PROTOBUF ?= otel/build-protobuf:0.4.0
+PROTOC := docker run --rm -u ${shell id -u} -v${PWD}:${PWD} -w${PWD} ${OTEL_DOCKER_PROTOBUF} --proto_path=${PWD}
```
Comment (mostynb) [inline Makefile:2]: I don't have a lot of experience with docker, what does this do? Does it generate the go protobuf bindings?

**Review by mostynb — COMMENTED** [review]
Thanks for the contribution.

If this is useful, I guess it might be worth adding a github action to run these tests?

File: Makefile (hunk 1 above)
Comment (bogdandrutu) [reply Makefile:2]: It generates the protobuf files using that docker image instead of requiring to have protoc and goproto plugin.

**Comment by mostynb** [comment]
Thanks for the contribution- merged with a minor fix for the test bindings generation, and made a new relase: v1.1.15.

I also landed a test-teardown flakiness fix after making the release- but I don't think that matters for downstream consumers.

**Pull request merged**